### DISCUSSION TITLE
Zeal Wallet - Changes requested

### DISCUSSION THREAD

**Discussion by Nicvaniek** [WalletConnect/walletconnect-monorepo#3624] [Developer Support] [answered]
Good day,

We submitted our explorer application last week and it was rejected due to the following reason: **_"Wallet Testing - Could not verify if testing has been done on this wallet. As we do not have available V2 metric data for Sign. Please test your integration with our test dapp and submit for approval if successful. Link: https://lab.web3modal.com/"_**

Could someone please assist me with what needs to change? We were able to successfully connect and sign on https://lab.web3modal.com/ (also included the testing instructions in the application). This is a browser injected wallet (not using Web3Wallet SDK).

See video below:

https://github.com/WalletConnect/walletconnect-monorepo/assets/14542853/8dc91df9-0820-467f-9add-9cdd92a214e6

**Comment by JonathanConn** [comment]
@Nicvaniek 

Happy to help out, your explorer submission lists the wallet as an injected wallet
In your demo, you used metamask to connect
If your wallet is an injected wallet, you would need to connect to our lab using your injected wallet, not metamask

Am I misunderstanding something? Let me know
-Jon

**Comment by JonathanConn** [comment]
If your wallet uses window.ethereum and the isMetaMask flag
You would have to disable the other extensions temporarily to ensure Zeal is the injected wallet that gets selected

Fun fact, there is an eip to fix multiple wallets using window.ethereum in the future
https://eips.ethereum.org/EIPS/eip-6963

**Reply by Nicvaniek** [reply]
Hi Jon, thanks for your response. I am not sure I understand the problem - I did exactly what you described above. Other extensions have been disabled so Zeal gets chosen as the injected wallet (we have isMetaMask=true and isZeal=true). If my understanding is correct, Zeal will only be an option on the modal once it is added to the explorer? We have made the necessary changes on Wagmi as well, so the latest version should include Zeal.

**Reply by JonathanConn** [reply]
> Hi Jon, thanks for your response. I am not sure I understand the problem - I did exactly what you described above. Other extensions have been disabled so Zeal gets chosen as the injected wallet (we have isMetaMask=true and isZeal=true). 

That is strange, could you record a demo for me, with your browsers dev console open, so I could assist better?

> If my understanding is correct, Zeal will only be an option on the modal once it is added to the explorer? 

Yes, once your submission is approved and published, it will be available into the explorer 

I'm sure we can figure this out for you quickly 

And a side note, it has to be Sign v2, not v1

**Reply by Nicvaniek** [reply]
Thanks Jon. Please see demo below:

https://github.com/WalletConnect/walletconnect-monorepo/assets/14542853/e996c2a0-1ec8-4027-8f47-9749266ccef5

**Reply by JonathanConn** [reply]
Looks great, we are on the right track
I noticed in your submission you marked Zeal as having Sign v1, not Sign v2
Can you confirm you are using Sign v2?

**Reply by Nicvaniek** [reply]
We're not using the WalletConnect SDK so not sure?

**Reply by JonathanConn** [reply]
What are you using?

I would recommend using this
https://docs.walletconnect.com/2.0/web3wallet/about

**Reply by JonathanConn** [reply]
Also, nice use of verify, it's very subtle but works nicely, well done

**Reply by Nicvaniek** [reply]
We just use RPC methods. The wallet supports `personal_sign`, `eth_signTypedData`, `eth_signTypedData_v3` and `eth_signTypedData_v4`

**Reply by JonathanConn** [reply]
Which WalletConnect packages are you using?

If it's easier, you can share the code snippet that contains your ProjectId, it would give me some more context as to what's going on

**Reply by Nicvaniek** [reply]
We're not using any WalletConnect packages - this might be where the confusion is coming in. Is it mandatory to use them to get the extension on the explorer?

**Reply by JonathanConn** [reply]
Yes, it is, here is a list requirements for the explore
https://walletconnect.com/guidelines

I would recommend implementing web3wallet
Here is a good example you can check out 
https://github.com/WalletConnect/web-examples/tree/main/wallets/react-web3wallet

If you do decide you want to join our explorer, and implement web3wallet, let us know and we can review your submission again

Best of luck, I'm rooting for Zeal

-Jon

**Reply by Nicvaniek** [reply]
I don't see any mention in the guidelines of it being mandatory to use the SDK? Just that it should support WalletConnect - which our extension does. Does the Metamask extension use the WalletConnect SDK? I also spoke to someone at BlockWallet who were able to get approved without using the SDK?

**Reply by JonathanConn** [reply]
MetaMask does use our sdk

I see, let me ask around and see what information I can get

**Reply by Nicvaniek** [reply]
Thanks!

**Reply by Nicvaniek** [reply]
I also have this Discord conversation - not sure if it helps:
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/14542853/69f51f70-2af8-49c3-8b44-b0e1c9a8d491)
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/14542853/b6f133c0-713d-4e14-93ab-84e1b3825c21)

**Reply by JonathanConn** [reply, marked as answer]
I actually just spoke with glitch and you are right, it's possible

I'll let the team know to approve you
You will receive an email confirming your project has been published
Give it about 30 minutes for the changes to show up in our explorer

Congratulations 🎉

**Reply by Nicvaniek** [reply]
Thanks Jon!!

**Comment by JonathanConn** [comment]
cc @JeroldG
